### ISSUE TITLE
`python310Packages.chainer` build failure on x86_64-linux as of `734b6c0a`

### ISSUE THREAD

**Issue by samuela** [NixOS/nixpkgs#199268]
## Issue description
Build of `python310Packages.chainer` failed on x86_64-linux as of 734b6c0a1c7745df0078736ae83888977fc2c028. This is currently breaking `python3Packages.einops`.

```
error: builder for '/nix/store/6iaprldms6m7y6xw1m0fd280bqspb8my-python3.10-chainer-7.8.1.post1.drv' failed with exit code 1;
       last 10 log lines:
       > adding 'chainer-7.8.1.dist-info/RECORD'
       > removing build/bdist.linux-x86_64/wheel
       > Finished executing setuptoolsBuildPhase
       > installing
       > Executing pipInstallPhase
       > /build/source/dist /build/source
       > Processing ./chainer-7.8.1-py3-none-any.whl
       > ERROR: Could not find a version that satisfies the requirement six>=1.9.0 (from chainer) (from versions: none)
       > ERROR: No matching distribution found for six>=1.9.0
       > 
       For full logs, run 'nix log /nix/store/6iaprldms6m7y6xw1m0fd280bqspb8my-python3.10-chainer-7.8.1.post1.drv'.
error: 1 dependencies of derivation '/nix/store/vy8r735qpn74v71k0jdpw0zzck1h14zz-python3.10-einops-0.4.1.drv' failed to build
```

[full build log](https://github.com/samuela/nixpkgs-upkeep/actions/runs/3382067127)

python3Packages.einops maintainers cc: @yl3dy
python310Packages.chainer maintainers cc: @hyphon81
Other cc: n/a

### Steps to reproduce
1. Checkout nixpkgs at commit 734b6c0a1c7745df0078736ae83888977fc2c028
2. Run `nix-build -A python310Packages.chainer`

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.0-1022-azure, Ubuntu, 20.04.5 LTS (Focal Fossa), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - nixpkgs: `/nix/store/bd6rjk9k0f06zvsispa5a0j0p5gh1bm0-source`
```

## Misc.
This issue was automatically generated by [nixpkgs-upkeep](https://github.com/samuela/nixpkgs-upkeep).
- [CI workflow](https://github.com/samuela/nixpkgs-upkeep/actions/runs/3382067127) that created this issue.
- Internal tags: 2465689711096795836 -4756057284282404257

**Comment by samuela** [comment]
not immediately clear to me why this is failing since our current version of six is 1.16.0:
https://github.com/NixOS/nixpkgs/blob/60974988b3e1e5d61341a5ef0357e7ae99fcf7f1/pkgs/development/python-modules/six/default.nix#L7-L9

and it hasn't been updated since 2021: https://github.com/NixOS/nixpkgs/commit/02d30961f489802ff24896c433abf9538731bb6d.

**Comment by samuela** [comment]
note to self: this is also breaking augmax

**Comment by fabaff** [comment]
Was fixed with #199315
